apps/src/labs/AppOptionsStore.ts

```diff
@@ -0,0 +1,100 @@
+// Load and parse app options data for a given level.
+import {AppOptions, ProjectType} from './types';
+
+export class AppOptionsStore {
+  currentAppOptions: AppOptions | null = null;
+
+  constructor(appOptions: AppOptions) {
+    this.loadAppOptions(appOptions);
+  }
+
+  // For now, we are just saving the data here from appOptions, which is sent on page load.
+  // In the future, we will have an api so we can hot reload levels.
+  loadAppOptions(appOptions: AppOptions) {
+    this.currentAppOptions = appOptions;
+  }
+
+  // We get the standalone app on the frontend because channels are created for project-backed levels
+  // without a project type.
+  // Open question: can we move this logic to the backend? All of this information comes from the backend.
+  getProjectType(): ProjectType | null {
```

Comment:
[this](https://github.com/code-dot-org/code-dot-org/blob/262eb4fe940b772a0d0e94b98af274eb707b4aa8/dashboard/legacy/middleware/channels_api.rb#L170) is the reason I added this logic here. I think we could improve our channel token creation to do this logic on the backend, but that's out of scope here. Curious for other's thoughts on this.

Reply:
+1 to doing this on the backend

Reply:
Agreed that we should be centralizing this kind of logic/list.

Reply:
I'm planning to move this to the backend as a follow-up, as part of creating an api to get/create channels/projects for a given level.